src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/ViewModels/DataSetMetaViewModels.cs

```diff
@@ -0,0 +1,209 @@
+
+    /// <summary>
+    /// The optimal number of decimal places that the indicator should use when displayed.
+    /// </summary>
+    public int? DecimalPlaces { get; init; }
+}
+
+/// <summary>
+/// A geographic level’s locations.
+/// </summary>
+public record LocationLevelMetaViewModel
+{
+    [JsonConverter(typeof(EnumToEnumValueJsonConverter<GeographicLevel>))]
+    public required GeographicLevel Level { get; init; }
+
+    /// <summary>
+    /// The locations belonging to this level.
+    /// </summary>
+    public required IReadOnlyList<LocationOptionMetaViewModel> Options { get; init; }
+}
+
+/// <summary>
+/// A geographic location e.g. a country, region, local authority, etc.
+/// </summary>
+[JsonConverter(typeof(LocationOptionMetaJsonConverter))]
+public abstract record LocationOptionMetaViewModel
+{
+    /// <summary>
+    /// The ID of the location.
+    /// </summary>
+    public required string Id { get; init; }
+
+    /// <summary>
+    /// The label of the location.
+    /// </summary>
+    public required string Label { get; init; }
+}
+
+public record LocationCodedOptionMetaViewModel : LocationOptionMetaViewModel
+{
+    /// <summary>
+    /// The code of the location.
+    /// </summary>
+    public required string Code { get; init; }
+}
+
+public record LocationLocalAuthorityOptionMetaViewModel : LocationOptionMetaViewModel
+{
+    /// <summary>
+    /// The ONS code of the local authority.
+    /// </summary>
+    public required string Code { get; init; }
+
+    /// <summary>
+    /// The old code (previously the LEA code) of the local authority.
+    /// </summary>
+    public required string OldCode { get; init; }
+}
+
+public record LocationProviderOptionMetaViewModel : LocationOptionMetaViewModel
+{
+    /// <summary>
+    /// The UKPRN (UK provider reference number) of the provider.
+    /// </summary>
+    public required string Ukprn { get; init; }
+}
+
+public record LocationRscRegionOptionMetaViewModel : LocationOptionMetaViewModel
+{
+}
+
+public record LocationSchoolOptionMetaViewModel : LocationOptionMetaViewModel
+{
+    /// <summary>
+    /// The URN (unique reference number) of the school.
+    /// </summary>
+    public required string Urn { get; init; }
+
+    /// <summary>
+    /// The LAESTAB (local authority establishment number) of the school.
+    /// </summary>
+    public required string LaEstab { get; init; }
+}
+
+/// <summary>
+/// A representation of a time period including a human-readable label.
+/// </summary>
+public record TimePeriodMetaViewModel
+{
+    [JsonConverter(typeof(EnumToEnumValueJsonConverter<TimeIdentifier>))]
+    public TimeIdentifier? Code { get; init; }
+
+    /// <summary>
+    /// The period that the time period relates to.
+    /// </summary>
+    public string Period { get; init; } = string.Empty;
+
+    /// <summary>
+    /// The time period in human-readable format.
+    /// </summary>
+    public string Label { get; init; } = string.Empty;
```

Comment:
Looks like these should be `required` and don't need empty strings as defaults

Reply:
This was me blindly copying the API schema [here](https://dfe-analytical-services.github.io/explore-education-statistics-api-docs/schemas/TimePeriodMetaViewModel/#properties) without thinking about it!